homeassistant/components/mqtt/discovery.py

```diff
@@ -118,151 +102,6 @@ def async_log_discovery_origin_info(
     )
 
 
-@callback
-def _replace_abbreviations(
-    payload: Any | dict[str, Any],
-    abbreviations: dict[str, str],
-    abbreviations_set: set[str],
-) -> None:
-    """Replace abbreviations in an MQTT discovery payload."""
-    if not isinstance(payload, dict):
-        return
-    for key in abbreviations_set.intersection(payload):
-        payload[abbreviations[key]] = payload.pop(key)
-
-
-@callback
-def _replace_all_abbreviations(discovery_payload: Any | dict[str, Any]) -> None:
```

Comment:
There are some good improvements here that we could split into another PR as well while this is being reworked